### PR TITLE
CSOT: Ignore wTimeoutMS 

### PR DESCRIPTION
**Pull request by vbabanin**
JAVA-4062

### CODE REVIEW

File: driver-core/src/main/com/mongodb/assertions/Assertions.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import java.util.function.Supplier;
 
 /**
- * <p>Design by contract assertions.</p> <p>This class is not part of the public API and may be removed or changed at any time.</p>
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/assertions/Assertions.java:28]: This statement seems to be a duplicate of the same statement on the line 36.

File: driver-core/src/main/com/mongodb/internal/async/function/RetryState.java [hunk 2]
```diff
@@ -203,7 +203,7 @@ private void doAdvanceOrThrow(final Throwable attemptException,
              */
             if (hasTimeoutMs() && !loopState.isLastIteration()) {
                 previouslyChosenException = createMongoTimeoutException(
-                        "MongoDB operation timed out during a retry attempt",
+                        "Retry attempt timed out.",
                         previouslyChosenException);
             }
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/internal/async/function/RetryState.java:208]: The retry could be triggered not only by transient errors in MongoDB operations but also within the scope of OIDC. Referring to it as a 'MongoDB operation' might lead to confusion among users.

File: driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java [hunk 3]
```diff
@@ -704,6 +707,40 @@ public void test10CustomTestWithTransactionUsesASingleTimeoutWithLock() {
         }
     }
 
+    /**
+     * Not a prose spec test. However, it is additional test case for better coverage.
+     */
+    @Tag("setsFailPoint")
+    @Test
+    @DisplayName("Should ignore wTimeoutMS of WriteConcern to initial and subsequent commitTransaction operations")
+    @Disabled //TODO JAVA-5425
```
Comment (vbabanin) [inline driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java:716]: An observation from [JAVA-5425](https://jira.mongodb.org/browse/JAVA-5425): TimeoutContext is null in subsequent commitTransaction operations.

File: driver-core/src/main/com/mongodb/internal/session/BaseClientSessionImpl.java [hunk 4]
```diff
@@ -228,4 +238,8 @@ protected TimeoutSettings getTimeoutSettings(final TransactionOptions transactio
                 .withMaxCommitMS(transactionOptions.getMaxCommitTime(MILLISECONDS))
                 .withTimeout(timeoutMS, MILLISECONDS);
     }
+
+    protected enum TransactionState {
```
Comment (rozza) [inline driver-core/src/main/com/mongodb/internal/session/BaseClientSessionImpl.java:242]: 👍

File: driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java (hunk 3 above)
Comment (rozza) [reply driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java:716]: Good catch

**Review by rozza — APPROVED** [review]
LGTM!

**Pull request merged**